pybox/utils.py

```diff
@@ -1,7 +1,20 @@
+from typing import Type, TypeVar
+
 __all__ = (
-    'singleton',
+    'Singleton',
+    'T'
 )
 
 
-def singleton(cls):
-    return cls()
+T = TypeVar('T')
+
+
+class Singleton:
+    def __init__(self, class_object: Type[T]) -> None:
```

Comment:
It's gonna be ok if instance attributes have been annotated like methods.
If the name of instance attributes will start with underscore, it will be awesome :)